pandas/core/internals/array_manager.py

```diff
@@ -869,6 +869,20 @@ def iset(
             self.arrays[mgr_idx] = value_arr
         return
 
+    def column_setitem(self, loc: int, idx: int | slice | np.ndarray, value) -> None:
+        """
+        Set values ("setitem") into a single column (not setting the full column).
+
+        This is a method on the ArrayManager level, to avoid creating an
+        intermediate Series at the DataFrame level (`s = df[loc]; s[idx] = value`)
+        """
+        arr = self.arrays[loc]
+        # create temporary SingleArrayManager without ref to use setitem implementation
+        mgr = SingleArrayManager([arr], [self._axes[0]])
+        new_mgr = mgr.setitem((idx,), value)
+        # update existing ArrayManager in-place
+        self.arrays[loc] = new_mgr.arrays[0]
```

Comment:
the "into" in the docstring suggests that the setting should occur into the existing array, so we shouldn't need to set a new array.  am i misunderstanding?

Reply:
Yes, that's right. But the `setitem` method does that (setting into the existing array), or more correctly _might_ do that.

So it certainly looks a bit confusing here, as it indeed seems that I am fully replacing the array for the column in question. 
But so if the `setitem` method changed the array inplace, this `self.arrays[loc] = new_mgr.arrays[0]` is assigning the original array again and thus not changing anything / a no-op. 
It's only if `setitem` actually created a new array (eg the setitem caused a dtype change), that this line is doing something relevant. 

In principle I could check if both arrays are identical, and only if that is not the case, do this `self.arrays[loc] = new_mgr.arrays[0]` (although it wouldn't actually change any behaviour, that might be more explicit)

Reply:
In general I find it also a bit confusing in our internal API that `setitem` (both the Manager and Block method) is a method that works in place (sometimes), but does return a manager/block.

Reply:
> In general I find it also a bit confusing in our internal API that setitem (both the Manager and Block method) is a method that works in place (sometimes), but does return a manager/block.

yah, id be open to a name change (separate PR) for that.  this is related to why i like setitem_inplace